sklearn/decomposition/pca.py

```diff
@@ -75,6 +80,8 @@ def _assess_dimension_(spectrum, rank, n_samples, n_features):
     spectrum_ = spectrum.copy()
     spectrum_[rank:n_features] = v
     for i in range(rank):
+        if spectrum_[i] < rcond:
```

Comment:
Can `spectrum_` values ever be negative here? If not, then the check above doesn't need the abs either, right?